src/Stratis.Bitcoin.Features.BlockStore.Tests/AddressIndexerTests.cs

```diff
@@ -123,8 +151,27 @@ public void CanIndexAddresses()
                 }
 
                 return new ChainedHeaderBlock(new Block(), header);
+            }
+
+            this.consensusManagerMock.Setup(x => x.GetBlockData(It.IsAny<uint256>())).Returns((uint256 hash) =>
+            {
+                return GetChainedHeaderBlock(hash);
             });
 
+            this.consensusManagerMock.Setup(x => x.GetBlockData(It.IsAny<List<uint256>>())).Returns((List<uint256> hashes) =>
+            {
+                return hashes.Select(h => GetChainedHeaderBlock(h)).ToArray();
+            });
+
+            this.consensusManagerMock.Setup(x => x.GetBlocksAfterBlock(It.IsAny<ChainedHeader>(), It.IsAny<int>(), It.IsAny<CancellationTokenSource>())).Returns((ChainedHeader header, int size, CancellationTokenSource token) =>
+            {
+                return headers.Select(h => GetChainedHeaderBlock(h.HashBlock)).ToArray();
```

Comment:
Just to double check, it is deliberate that `GetBlocksAfterBlock` returns the blocks associated with the entire contents of `headers` without any filtering?

Reply:
For this test case, yes.

Reply:
Either way, I've updated it to be more intuitive.